webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py

```diff
@@ -0,0 +1,258 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import unicode_literals
+
+from past.utils import old_div
+from builtins import range
+from builtins import object
+import helpers
+import constants
+import copy
+
+class LoadProfile(object):
+
+    def __init__(self, name):
+
+        self.name = name
+        self.annual_load = {}
+        self.avg_weekday = {}
+        self.avg_weekend = {}
+        self.peak_shape = {}
+        self.peak_isweekday = {}
+        self.flexload_added = False
+
+
+    def is_empty(self):
+
+        if self.annual_load == {}:
+            if self.avg_weekday == {}:
+                if self.avg_weekend == {}:
+                    if self.peak_shape == {}:
+                        if self.peak_isweekday == {}:
+                            return True

```

Comment:
Did we write this class?
We need to re-write this in favor of using `not dict_object_im_evaluating` instead of `self.dict_object_im_evaluating == {}`.
Also please don't nest the conditionals like this and we need to `return` on all code paths.

Reply:
this is code not written by us. I can get in touch with Robbie and see if it ok to make these changes, but I know we will be receiving a new cba tool relatively soon